From bb8724ffe24a82ce22c7ccc33c97e2c5bbf74aca Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <heinrich.schuchardt@canonical.com>
Date: Fri, 2 Sep 2022 13:40:57 +0200
Subject: [PATCH 1/1] commands/efi/lsefisystab: Short text for
 EFI_CONFORMANCE_PROFILES_TABLE

The EFI_CONFORMANCE_PROFILES_TABLE_GUID is used for a table of GUIDs for
conformance profiles (cf. UEFI specification 2.10,
4.6.5 EFI_CONFORMANCE_PROFILE_TABLE).

The lsefisystab command is used to display installed EFI configuration
tables. Currently it only shows the GUID but not a short text for the
table.

Provide a short text for the EFI_CONFORMANCE_PROFILES_TABLE_GUID.

Signed-off-by: Heinrich Schuchardt <heinrich.schuchardt@canonical.com>
---
 grub-core/commands/efi/lsefisystab.c | 1 +
 include/grub/efi/api.h               | 5 +++++
 2 files changed, 6 insertions(+)

diff --git a/grub-core/commands/efi/lsefisystab.c b/grub-core/commands/efi/lsefisystab.c
index f24906be6..90b9c93a5 100644
--- a/grub-core/commands/efi/lsefisystab.c
+++ b/grub-core/commands/efi/lsefisystab.c
@@ -37,6 +37,7 @@ static const struct guid_mapping guid_mappings[] =
   {
     { GRUB_EFI_ACPI_20_TABLE_GUID, "ACPI-2.0"},
     { GRUB_EFI_ACPI_TABLE_GUID, "ACPI-1.0"},
+    { GRUB_EFI_CONFORMANCE_PROFILES_TABLE_GUID, "CONFORMANCE PROFILES TABLE"},
     { GRUB_EFI_CRC32_GUIDED_SECTION_EXTRACTION_GUID,
       "CRC32 GUIDED SECTION EXTRACTION"},
     { GRUB_EFI_DEBUG_IMAGE_INFO_TABLE_GUID, "DEBUG IMAGE INFO"},
diff --git a/include/grub/efi/api.h b/include/grub/efi/api.h
index 1ef404622..634677d9c 100644
--- a/include/grub/efi/api.h
+++ b/include/grub/efi/api.h
@@ -294,6 +294,11 @@
     { 0x9c, 0x10, 0x8b, 0x97, 0xa8, 0x3f, 0xfd, 0xfa } \
   }
 
+#define GRUB_EFI_CONFORMANCE_PROFILES_TABLE_GUID \
+ { 0x36122546, 0xf7e7, 0x4c8f, \
+   { 0xbd, 0x9b, 0xeb, 0x85, 0x25, 0xb5, 0x0c, 0x0b } \
+ }
+
 #define GRUB_EFI_DEBUG_IMAGE_INFO_TABLE_GUID \
   { 0x49152e77, 0x1ada, 0x4764, \
     { 0xb7, 0xa2, 0x7a, 0xfe, 0xfe, 0xd9, 0x5e, 0x8b } \
-- 
2.37.2

